### ISSUE TITLE
PoC: Generate a 1ES managed image based on one of the base image definitions in helix-machines

### ISSUE THREAD

**Issue by riarenas** [dotnet/dnceng#613]
Test what it would look like to deploy a 1es managed image that generates one of the images we currently generate using the image factory CLI.

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements
### Release Note Description

**Comment by riarenas** [comment]
I started by manually attempting to recreate one of the VS images. Specifically the [helix-server-datacenter-2022-vs2-2022-preview-enterprise](https://dnceng.visualstudio.com/internal/_git/dotnet-helix-machines?path=%2Fbase-images%2FVS2022Preview%2FHelix-Server-Datacenter-2022-VS2022-Preview-Enterprise.json&_a=contents&version=GBmain) image. 

I manually created a [1es image resource in the portal](https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/84a65c9a-787d-45da-b10a-3a1cefce8060/resourceGroups/riarenas-image-tests/providers/Microsoft.CloudTest/images/riarenas-Helix-Server-Datacenter-2022-VS2022-Preview/overview) based on the properties of this image. I was able to mostly specify the artifacts as they were in the json definition, except I had to replace all uppercase property names with lowercase.

Once the Azure resource is created, Image Factory is then called asynchronously to provision the internal image. The portal mentions this could take 2-5 hours. While the image is generating, the ARM template for the resource includes the image factory status:

```
        "subscriptionType": "Azure",
        "imageType": "Managed",
        "provisioningState": "Updating",
        "requestId": "de3c2285-0c81-475b-9702-6be31e6728bf"
```

so during deployments we have to account that it's not enough to deploy the resource to ensure the image is valid and works. 

We should look more into the image validation features: https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/1es-managed-images/validation which could mean we don't wait for the images to be ready, and instead use the validation pipelines to make sure everything went well after the fact.
